cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -102,12 +102,26 @@ public void createCollection(String dbName, String collectionName, long dataExpi
         BasicDBObject keys;
         IndexOptions options;
         try {
-            keys = new BasicDBObject()
-                .append("_id.entityId", 1)
-                .append("_id.entityType", 1)
-                .append("_id.attrName", 1)
-                .append("_id.resolution", 1)
-                .append("_id.origin", 1);
+            keys = new BasicDBObject();
+                switch(dataModel) {
+                case DMBYSERVICEPATH:
+                    keys.append("_id.entityId", 1)
+                        .append("_id.entityType", 1)
+                        .append("_id.attrName", 1)
+                        .append("_id.origin", 1)
+                        .append("_id.resolution", 1);
```

Comment:
Why this change in the order of keys instead of keeping the original one? I mean:

```suggestion
                        .append("_id.resolution", 1);
                        .append("_id.origin", 1)
```